api/app/routers/pteams.py

```diff
@@ -827,6 +838,35 @@ def remove_specified_group_references_from_pteamtag(db, pteamtag, group):
         db.add(pteamtag)
 
 
+@router.post("/{pteam_id}/upload_sbom_file", response_model=List[schemas.ExtTagResponse])
+def upload_pteam_sbom_file(
+    pteam_id: UUID,
+    file: UploadFile,
+    group: str = Query("", description="name of group(repository or product)"),
+    force_mode: bool = Query(False, description="if true, create unexist tags"),
+    current_user: models.Account = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """
+    upload sbom file
+    """
+    pteam = validate_pteam(db, pteam_id, on_error=status.HTTP_404_NOT_FOUND)
+    assert pteam
+    check_pteam_membership(db, pteam_id, current_user.user_id, on_error=status.HTTP_403_FORBIDDEN)
+    _check_file_extention(file, ".json")
+    _check_empty_file(file)
+    jdata = _json_load(file.file)
+
+    try:
+        SbomFormat.from_jsondata(jdata)  # check sbom format
```

Comment:
将来の拡張を考えるなら、一応戻り値の判定もしておいた方がいいと思います (このコードを見た人が迷わないように)

```suggestion
        if SbomFormat.from_jsondata(jdata) == SbomFormat.CDX:
            components = gen_cdx_components(jdata)
```